### PR TITLE
Functions for using named cronjobs

### PR DESCRIPTION
**Pull request by KoronGIT**
Node-cron is used.

Functions:

create
remove
addTaskToCron
removeTaskFromCron

Usage Example:

```
import { useRebar } from '@Server/index.js';

const Rebar = useRebar();
const cronJob = Rebar.useCronJob();

export function task1() {
    alt.log('Task 1');
}

export function task2() {
    alt.log('Task 2');
}

declare global {
    export interface CronJobs {
        'my-cronjob': {
            cronExpression: string;
            tasks: any[];
        };
    }
}

cronJob.create('my-cronjob', '*/5 * * * * *', [task1]);
cronJob.addTaskToCron('my-cronjob', task2);
```

### CODE REVIEW

File: src/main/server/cronjob/index.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+        return true;
+    }
+
+    function remove(jobName: string) {
+        if (registeredJobs.has(jobName)) {
+            const { job } = registeredJobs.get(jobName);
+            job.stop();
+            registeredJobs.delete(jobName);
+        }
```
Comment (Stuyk) [inline src/main/server/cronjob/index.ts:38]:
```suggestion
        if (!registeredJobs.has(jobName)) {
            return;
        }
        
        const { job } = registeredJobs.get(jobName);
        job.stop();
        registeredJobs.delete(jobName);
```

File: src/main/server/cronjob/index.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    function addTaskToCron<K extends keyof CronJobs>(jobName: K, task: () => void) {
+        if (registeredJobs.has(jobName)) {
+            const { job, cronExpression, tasks } = registeredJobs.get(jobName);
+            job.stop();
+
+            tasks.push(task);
+
+            create(jobName, cronExpression, tasks, true);
+        }
```
Comment (Stuyk) [inline src/main/server/cronjob/index.ts:49]:
```suggestion
        if (!registeredJobs.has(jobName)) {
            return;
        }
        
        const { job, cronExpression, tasks } = registeredJobs.get(jobName);
        job.stop();

        tasks.push(task);
        create(jobName, cronExpression, tasks, true);
```

File: src/main/server/cronjob/index.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+            job.stop();
+
+            const index = tasks.indexOf(task);
+            if (index > -1) {
+                tasks.splice(index, 1);
+            }
+
+            create(jobName, cronExpression, tasks, true);
+        }
+    }
```
Comment (Stuyk) [inline src/main/server/cronjob/index.ts:64]:
```suggestion
    function removeTaskFromCron<K extends keyof CronJobs>(jobName: K, task: () => void) {
        if (!registeredJobs.has(jobName)) {
            return;
        }
        
        const { job, cronExpression, tasks } = registeredJobs.get(jobName);
        job.stop();

        const index = tasks.indexOf(task);
        if (index > -1) {
            tasks.splice(index, 1);
        }

        create(jobName, cronExpression, tasks, true);
    }
```

**Review by Stuyk — CHANGES_REQUESTED** [review]

**Comment by Stuyk** [comment]
Additionally, could use a documentation page on this.

**Comment by KoronGIT** [comment]
> Additionally, could use a documentation page on this.

I added documentation

File: docs/useRebar/useCronJob.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
+---
+order: 930
+---
+
+# useCronJob
+
+Allow plugins to create cronjobs.
```
Comment (Stuyk) [inline docs/useRebar/useCronJob.md:7]:
```suggestion
This API allows you to create timed events that happen at different intervals based on server time. Such as every 5th minute of the hour, every hour, etc. 
```

**Pull request merged**